File: WebApiSpark/src/main/java/Model/Location.java
package Model;


import Util.DBHelper;
import Util.DatabaseValues;

import java.util.List;

public class Location extends IncidentElement
{
    public Location ( )
    {
        this (
                "",
                "",
                "",
                "",
                ""

        );
    }

    public Location (
            String locationID,
            String campusID,
            String buildingName,
            String roomNumber,
            String department
    ) {
        super (
                DatabaseValues.Table.LOCATION,
                new DatabaseValues.Column[]
                {
                        DatabaseValues.Column.LOCATION_ID,
                        DatabaseValues.Column.CAMPUS_ID,
                        DatabaseValues.Column.BUILDING_NAME,
                        DatabaseValues.Column.ROOM_NUMBER,
                        DatabaseValues.Column.DEPARTMENT,
                }
        );

        updateAttributeValue(
                DatabaseValues.Column.LOCATION_ID,
                locationID
        );

        updateAttributeValue(
                DatabaseValues.Column.CAMPUS_ID,
                campusID
        );

        updateAttributeValue(
                DatabaseValues.Column.BUILDING_NAME,
                buildingName
        );

        updateAttributeValue(
                DatabaseValues.Column.ROOM_NUMBER,
                roomNumber
        );

        updateAttributeValue(
                DatabaseValues.Column.DEPARTMENT,
                department
        );
    }

    @Override
    public String toSearchString() {

        // HACK
        Campus[] campusList = DBHelper.getCampus();
        int index = Integer.parseInt(this.getAttributeValue( DatabaseValues.Column.CAMPUS_ID));
        String campusString = campusList[index - 1].getAttributeValue( DatabaseValues.Column.CITY);

        return campusString + " " +
                this.getAttributeValue( DatabaseValues.Column.BUILDING_NAME ) + " " +
                this.getAttributeValue( DatabaseValues.Column.ROOM_NUMBER ) + " " +
                this.getAttributeValue( DatabaseValues.Column.DEPARTMENT );
    }
}

File: IRSClientAngular/src/app/model/location.ts
export class Location {
  LOCATION_ID: number;
  CAMPUS_ID: number;
  BUILDING_NAME: string;
  ROOM_NUMBER: number;
  DEPARTMENT: string;
  CITY: string;
  ADDRESS: string;
  
  /*

  Location = "{columValue:{ROOM_NUMBER:\"SURR-301\",CAMPUS_ID:\"1\",BUILDING_NUMBER:\"23\",DEPARTMENT:\"Sci\"}}"
  Incident = "{ColumnValue:{}, Location, Staff}"


[8:54] 
  constructor ( 
      id: number , 
      campus: string ,
      buildingNumber: string ,
      roomNumber: number ,
      department: string
  ) {
      this.id = id;
      this.campus = campus;
      this.buildingNumber = buildingNumber;
      this.roomNumber = roomNumber;
      this.department = department;
  }*/
}